arbnode/seq_coordinator.go

```diff
@@ -525,12 +525,18 @@ func (c *SeqCoordinator) update(ctx context.Context) time.Duration {
 	if c.prevChosenSequencer == c.config.MyUrl {
 		return c.updatePrevKnownChosen(ctx, chosenSeq)
 	}
-	if chosenSeq != c.config.MyUrl && chosenSeq != c.prevChosenSequencer {
+	chosenSeqChanged := chosenSeq != c.prevChosenSequencer
+	wrongForwardingTarget := c.sequencer != nil && c.sequencer.ForwardTarget() != chosenSeq
+	if chosenSeq != c.config.MyUrl && (chosenSeqChanged || wrongForwardingTarget) {
+		if chosenSeqChanged {
+			log.Info("chosen sequencer changed", "chosen", chosenSeq)
+		} else {
+			log.Info("attempting to reconnect to chosen sequencer", "chosen", chosenSeq)
+		}
 		if c.sequencer != nil {
 			c.sequencer.ForwardTo(chosenSeq)
```

Comment:
Another option: have ForwardTo return an error when it fails, and if it does - set c.prevchosensequencer to empty string.
This will make this case equal to what happened if redis doesn't show any live sequencer active, which is nice.

Reply:
Good idea, updated :)